src/hooks/useBalances.ts

```diff
@@ -3,22 +3,52 @@ import isEqual from 'lodash/isEqual'
 import { type SafeBalanceResponse } from '@safe-global/safe-gateway-typescript-sdk'
 import { useAppSelector } from '@/store'
 import { selectBalances } from '@/store/balancesSlice'
+import type { HiddenAssetsOnChain } from '@/store/hiddenAssetsSlice'
+import useHiddenAssets from './useHiddenAssets'
+import { safeFormatUnits, safeParseUnits } from '@/utils/formatters'
+import { BigNumber } from 'ethers'
 
-const useBalances = (): {
+const removeHiddenBalances = (
+  balances: SafeBalanceResponse,
+  hiddenAssets: HiddenAssetsOnChain,
+): SafeBalanceResponse => {
+  const fiatTotalVisible = safeFormatUnits(
+    balances.items
+      .reduce((acc, balanceItem) => {
+        if (typeof hiddenAssets?.[balanceItem.tokenInfo.address] !== 'undefined') {
```

Comment:
I still think we shouldn't touch the total balance. What did backend and mobile say to this?

Sidenote: we use `typeof === 'undefined'` for entirely undeclared variables, for regular null/undefined-values you can simply do `== null` or `=== undefined`.

Reply:
This is the last discussion point where I want product to have the last call. If @johannesmoormann decides to leave it untouched it is a very quick change. 

The decision was made based on the fact that there are spam tokens which are part of liquidity pools leading to price oracles returning a USD value for them. This could cloud the real total USD balance / value of the Safe.

Reply:
I'm talking about the solution with summing up fiat values on the client side.
Moving this logic to `useVisibleBalances` as discussed on Slack makes it slightly better.